lighty-core/lighty-codecs/src/main/java/io/lighty/codecs/SerializeIdentifierCodec.java

```diff
@@ -73,10 +100,60 @@ public YangInstanceIdentifier serialize(final String identifier) {
                     builder.node(buildNodeWithValue(leafListSchemaNode, args));
                 }
             } else {
+                // If node does not exist in schema tree, just append the name and do not dive deeper
                 builder.node(qName);
             }
         }
         return builder.build();
+
+    }
+
+    /**
+     * Finds the module from loaded schema context based on name and revision.
+     * If multiple revisions of module are present in the schema, and no specific revision was provided
+     * we return the module with the latest revision.
+     *
+     * @param moduleName     name of the module to find
+     * @param revisionString revision of the module
+     * @return found module
+     */
+    private Module findModule(String moduleName, String revisionString) {
+        Optional<Revision> requestedRevision;
+        try {
+            requestedRevision = Revision.ofNullable(revisionString);
+        } catch (DateTimeException e) {
+            throw new IllegalStateException(e + "--> Wrongly formatted revision provided" + revisionString);
+        }
+        Collection<? extends Module> modules = this.schemaContext.findModules(moduleName);
+
+        if (modules.isEmpty()) {
+            throw new IllegalStateException("Module with name " + moduleName + " not found");
+        }
+
+        Optional<? extends Module> foundModule = Optional.empty();
+
+        if (requestedRevision.isPresent()) {
+            Optional<Revision> finalRequestedRevision = requestedRevision;
+            foundModule = modules.stream()
+                    .filter(m -> m.getRevision().isPresent())
+                    .filter(m -> m.getRevision().get().equals(finalRequestedRevision.get()))
+                    .findFirst();
+        }
+
+        if (modules.size() > 1 && foundModule.isEmpty()) {
+            LOG.debug("Multiple revisions of module {} found: ", moduleName);
+            modules = modules.stream()
+                    .filter(m -> m.getRevision().isPresent())
+                    .peek(m -> LOG.debug("\t Revision: {} ", m.getRevision().get().toString()))
+                    .sorted((m1, m2) -> m2.getRevision().get().compareTo(m1.getRevision().get()))
+                    .collect(Collectors.toList());
+            LOG.debug("Using latest revision");
```

Comment:
Use WARN here and add more information - it is potentially dangerous to suggest what may user want (to return latest revision)